zebra-network/src/peer/client/tests.rs

```diff
@@ -1,3 +1,153 @@
-//! Tests for the [`Client`] part of peer connections
+//! Tests for the [`Client`] part of peer connections, and some test utilities for mocking
+//! [`Client`] instances.
 
 mod vectors;
+
+use futures::channel::{mpsc, oneshot};
+
+use crate::{
+    peer::{error::SharedPeerError, CancelHeartbeatTask, Client, ClientRequest, ErrorSlot},
+    protocol::external::types::Version,
+};
+
+/// A handle to a mocked [`Client`] instance.
+pub struct MockedClientHandle {
+    request_receiver: Option<mpsc::Receiver<ClientRequest>>,
+    shutdown_receiver: Option<oneshot::Receiver<CancelHeartbeatTask>>,
+    error_slot: ErrorSlot,
+    version: Version,
+}
+
+impl MockedClientHandle {
+    /// Create a new mocked [`Client`] instance, returning it together with a handle to track it.
+    pub fn new(version: Version) -> (Self, Client) {
+        let (shutdown_sender, shutdown_receiver) = oneshot::channel();
+        let (request_sender, request_receiver) = mpsc::channel(1);
+        let error_slot = ErrorSlot::default();
+
+        let client = Client {
+            shutdown_tx: Some(shutdown_sender),
+            server_tx: request_sender,
+            error_slot: error_slot.clone(),
+            version,
+        };
+
+        let handle = MockedClientHandle {
+            request_receiver: Some(request_receiver),
+            shutdown_receiver: Some(shutdown_receiver),
+            error_slot,
+            version,
+        };
+
+        (handle, client)
+    }
+
+    /// Gets the peer protocol version associated to the [`Client`].
+    pub fn version(&self) -> Version {
+        self.version
+    }
+
+    /// Checks if the [`Client`] instance has not been dropped, which would have disconnected from
+    /// the peer.
+    pub fn is_connected(&mut self) -> bool {
+        let receive_result = self
+            .shutdown_receiver
+            .as_mut()
+            .expect("shutdown receiver has been dropped")
+            .try_recv();
+
+        match receive_result {
+            Ok(None) => true,
+            Ok(Some(CancelHeartbeatTask)) | Err(oneshot::Canceled) => false,
+        }
+    }
+
+    /// Drops the shutdown receiver endpoint.
+    pub fn drop_shutdown_receiver(&mut self) {
+        let _ = self
+            .shutdown_receiver
+            .take()
+            .expect("request receiver endpoint has already been dropped");
+    }
+
+    /// Closes the request receiver endpoint.
+    pub fn close_request_receiver(&mut self) {
```

Comment:
```suggestion
    /// Closes the outbound peer request receiver endpoint.
    pub fn close_outbound_peer_request_receiver(&mut self) {
```

Reply:
Actually, maybe this is unclear - what I mean is "the receiver endpoint for outbound requests to the mocked peer".

Reply:
Done, I tried to phrase it to be clearer. Let me know what you think.